spv/bump_validation.go

```diff
@@ -0,0 +1,82 @@
+package spv
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/bitcoin-sv/go-paymail/beef"
+	"github.com/libsv/go-bt/v2"
+)
+
+func ensureAncestorsArePresentInBump(tx *bt.Tx, dBeef *beef.DecodedBEEF) error {
+	ancestors, err := findMinedAncestors(tx, dBeef.Transactions)
+	if err != nil {
+		return err
+	}
+
+	for _, tx := range ancestors {
+
+		if !existsInBumps(tx, dBeef.BUMPs) {
+			return errors.New("invalid BUMP - input mined ancestor is not present in BUMPs")
+		}
+	}
+
+	return nil
+}
+
+func findMinedAncestors(tx *bt.Tx, ancestors []*beef.TxData) (map[string]*beef.TxData, error) {
+	am := make(map[string]*beef.TxData)
+
+	for _, input := range tx.Inputs {
+		err := findMinedAncestorsForInput(input, ancestors, am, 0)
+
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return am, nil
+}
+
+func findMinedAncestorsForInput(input *bt.Input, ancestors []*beef.TxData, ma map[string]*beef.TxData, depth uint) error {
+	if depth > 128 {
```

Comment:
There is something like that in documentation or is it our check?

Reply:
It's arbitrarily chosen value for max recursion depth. It's simple protection, not perfect, against cyclic graphs and infinite loops. 
IMO it's good enough for now but probably we would work on it later